lib/curl_threads.h

```diff
@@ -37,16 +37,19 @@
 #  define curl_mutex_t           CRITICAL_SECTION
 #  define curl_thread_t          HANDLE
 #  define curl_thread_t_null     (HANDLE)0
+/* The Windows init/destroy macros are set to return 0 (success) even though
+   the respective Windows API functions do not return values. That makes them
+   behave the same as pthreads init/destroy which return 0 on success. */
 #  if !defined(_WIN32_WINNT) || !defined(_WIN32_WINNT_VISTA) || \
       (_WIN32_WINNT < _WIN32_WINNT_VISTA) || \
       (defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR))
-#    define Curl_mutex_init(m)   InitializeCriticalSection(m)
+#    define Curl_mutex_init(m)   (InitializeCriticalSection(m), 0)
 #  else
-#    define Curl_mutex_init(m)   InitializeCriticalSectionEx(m, 0, 1)
+#    define Curl_mutex_init(m)   (InitializeCriticalSectionEx(m, 0, 1), 0)
```

Comment:
This one actually returns a `BOOL`: https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initializecriticalsectionex

Reply:
Oops. Thanks, fixed.